en/test_and_ci/integration_testing_mavsdk.md

```diff
@@ -2,8 +2,8 @@
 
 PX4 can be tested end to end to using integration tests based on [MAVSDK](https://mavsdk.mavlink.io).
 
-The tests are primarily developed against SITL for now and run in continuous integration (CI).
-However, they are meant to generalize to real tests eventually.
+The tests are primarily developed against SITL and run in continuous integration (CI).
+However, the intent is to generalise them for any platform/hardware in future.
```

Comment:
@julianoes So it looks like here we explain how to run all tests, and some background on the test suite. Other things I might reasonably want to do (and which are covered in the ROS tests)
1. Run just one of the tests
1. Find out what tests can be run - ie.e names and what they cover
1. Write my own new test and make sure it is always run in CI. 
1. Understand when I should use MAVSDK for writing a test - vs writing "SITL unit tests".

Can you advise on any of that?

Reply:
2. Find out what tests there are by running them all, they are nicely listed: `test/mavsdk_tests/mavsdk_test_runner.py test/mavsdk_tests/configs/sitl.json --speed-factor 10`

1. Run just one case and/or model: `test/mavsdk_tests/mavsdk_test_runner.py test/mavsdk_tests/configs/sitl.json --speed-factor 10 --model tailsitter --case 'Fly square Multicopter Missions including RTL'`

3. Add tests in https://github.com/PX4/PX4-Autopilot/tree/master/test/mavsdk_tests

4. Unit tests should be used to test new (or changed) functionality around a small "unit" e.g. a class or a module.
Integration tests should be used to test an overall behaviour to make sure the units play together. But an integration test can probably never test each and every code path because the state space is just too big.